[FLINK-10569] [tests] Get rid of Scheduler from valid tests

TisonKun opened the pull request:
## What is the purpose of the change

Legacy class `Scheduler` and `Instance` are still used in some valid tests like `ExecutionGraphRestartTest`. We should replace them with FLIP-6 schedule mode. The best way I can find is use `TestingSlotProvider` or `ProgrammedSlotProvider`.

**This is now part of FLINK-10569 and I'm going to port tests commit by commit. They are a bit orthogonal so I'm also glad to see we get things done in several pull requests.**

## Brief change log

- `Scheduler` -> `TestingSlotProvider`
  - `PointwisePatternTest`
  - `ExecutionGraphCheckpointCoordinatorTest`
  - `ExecutionGraphConstructionTest`
  - `ExecutionGraphTestUtils#getExecutionVertex`
  - `RestartPipelinedRegionStrategyTest`
  - `VertexSlotSharingTest`
  - `RescalePartitionerTest`
  - `ExecutionGraphMetricsTest`
  - `ExecutionVertexSchedulingTest`
  - `ExecutionVertexCancelTest`
  - `ExecutionGraphDeploymentTest`
  - `ExecutionGraphRescalingTest`

- Removed
  - `SchedulerSlotSharingTest` is covered by `SlotPoolCoLocationTest` and `SlotPoolSlotSharingTest`
  - `SchedulerTest` is covered by `SlotPoolRpcTest`

## Verifying this change

This change is a trivial rework / code cleanup without any more test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (**no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (**no**)
  - The serializers: (**no**）
  - The runtime per-record code paths (performance sensitive): (**no**）
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (**no**）
  - The S3 file system connector: (**no**）

## Documentation

  - Does this pull request introduce a new feature? (**no**）
  - If yes, how is the feature documented? (**not applicable**）

TisonKun commented:
>03:14:22.321 [ERROR] Tests run: 10, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 3.189 s <<< FAILURE! - in org.apache.flink.client.program.rest.RestClusterClientTest
03:14:22.322 [ERROR] testSendIsNotRetriableIfHttpNotFound(org.apache.flink.client.program.rest.RestClusterClientTest)  Time elapsed: 0.043 s  <<< ERROR!
java.net.BindException: Address already in use

Travis failed as above. I guess it is about another issue.

TisonKun commented on flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/VertexSlotSharingTest.java:
```diff
@@ -90,12 +90,12 @@ public void testAssignSlotSharingGroup() {
 			// verify that the vertices are all in the same slot sharing group
-			SlotSharingGroup group1 = null;
-			SlotSharingGroup group2 = null;
+			SlotSharingGroup group1;
```
It is accidental to be in the same commit. I am not quite sure if I have to revert it. What is the most convenient commits layout for merging a pull request and how does flink deal with such "code cleaning" works? @zentol

zentol replied:
I would revert this as I wouldn't even categorize this as code-cleanup. Whether to keep these uninitialized or not is more or less personal preference.

TisonKun replied:
Gotcha.

TisonKun replied:
Revert while rebasing.

TisonKun commented:
cc @tillrohrmann 

I'd like to finalize this pr. Since the rest of valid usages of `Scheduler` are in `ExecutionGraphRestartTest` and `FailoverRegionTest`, which is coupled with `Instance`. Later I would diagnose how to port Instance.

zentol commented on flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/ExecutionGraphRestartTest.java:
```diff
@@ -155,21 +153,12 @@ public void testRestartAutomatically() throws Exception {
+		assertEquals(JobStatus.CREATED, executionGraph.getState());
 
+		executionGraph.scheduleForExecution();
 		assertEquals(JobStatus.RESTARTING, executionGraph.getState());
```
why is the EG in a restarting state? Previously this was caused by killing the taskmanager, but what is causing this now?

TisonKun replied:
This is because the SlotProvider returns non-completed future. And eg is configured as not allow queued scheduling. Thus when scheduleForExecutor, it causes `IllegalArgumentException("The slot allocation future has not been completed yet.")` and then failing and then restarting.

But it is tricky. Maybe I should set allow queued scheduling and fail a future to emulate a real failure.

zentol replied:
If we fail the future right away the scheduling mode should be irrelevant, correct? It should fail in any case and restart the job.

TisonKun replied:
Sure.

zentol commented:
I would propose to merge part of this PR, namely all tests that are finished in terms of porting. The majority of ports in this PR are rather straight-forward and can be pushed as one commit.
(as all they do is replace `new Scheduler(TestingUtils.defaultExecutionContext())` with `new TestingSlotProvider(ignored -> new CompletableFuture<>())`.)

For the others I would propose to group them in PRs based on the required changes. Exceptions should be made for tests that require substantial changes to ensure steady progress.

TisonKun commented:
@zentol I've rebased the commits, excluded `ExecutionGraphRestartTest` and fixed conflicts with master branch. If I understand it correctly, it is a green status for you to merge this part of the whole porting job.

I'll keep working to finally get rid of `Instance` and `Scheduler`.

TisonKun commented:
travis fails as below, which I think is irrelevant to this pr.

>20:58:02.307 [WARNING] The requested profile "include-kinesis" could not be activated because it does not exist.
MVN exited with EXIT CODE: 0.
Trying to KILL watchdog (12036).
./tools/travis_mvn_watchdog.sh: line 269: 12036 Terminated              watchdog
PRODUCED build artifacts.
build_info  mvn-1.log  mvn-2.log  mvn.out
COMPRESSING build artifacts.
build_info
container.log
mvn-1.log
mvn-2.log
mvn.out
Uploading to transfer.sh
>
>
> No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#Build-times-out-because-no-output-was-received
>
> The build has been terminated
